### PR TITLE
Extending BaseImaging to track dtype to match SpikeInterface (#132)

### PR DESCRIPTION
**Pull request by joannercsheppard**
BaseImaging extended to store dtype instead of inferring from samples so now BaseProcessor uses the dtype in constructor.

### CODE REVIEW

File: src/photon_mosaic/core/baseimaging.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ class BaseImaging(BaseExtractor, TimeSeries):
     The `_main_ids` attribute is used here for multi-plane imaging objects.
     """
 
-    def __init__(self, sampling_frequency: float, shape: tuple | list | ArrayLike):
+    def __init__(self, sampling_frequency: float, shape: tuple | list | ArrayLike, dtype: DTypeLike | None = None):
```
Comment (alejoe91) [inline src/photon_mosaic/core/baseimaging.py:22]:
```suggestion
    def __init__(self, sampling_frequency: float, shape: tuple | list | ArrayLike, dtype: DTypeLike):
```
I think we should enforce `dtype` to be not `None`

**Comment by alejoe91** [comment]
@joannercsheppard can you fix linting test? My suggested change means that we have to add `dtype` to a bunch of places :)

File: src/photon_mosaic/core/baseimaging.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ class BaseImaging(BaseExtractor, TimeSeries):
     The `_main_ids` attribute is used here for multi-plane imaging objects.
     """
 
-    def __init__(self, sampling_frequency: float, shape: tuple | list | ArrayLike):
+    def __init__(self, sampling_frequency: float, shape: tuple | list | ArrayLike, dtype: DTypeLike):
```
Comment (alessandrofelder) [inline src/photon_mosaic/core/baseimaging.py]: Should `dtype` be `None` by default or do we expect users to explicitly set it to `None` if they want to infer from data? Maybe I misunderstood 🤔

File: src/photon_mosaic/preprocessing/tests/test_basepreprocessor.py [hunk 3]
```diff
@@ -37,6 +38,24 @@ def test_init_custom_dtype(self, imaging):
...
+        assert reg.get_dtype() == np.dtype(np.float32)
+
+    def test_get_dtype_defaults_to_parent_dtype(self, imaging):
+        # Verify that the default dtype is inherited from the parent imaging object
+        reg = BasePreprocessor(imaging)
+        assert reg.get_dtype() == imaging.get_dtype()
+
+    def test_get_dtype_with_explicit_dtype_does_not_sample_parent_data(self):
+        # An imaging object with no epochs has no data to sample. If get_dtype()
+        # tried to infer the dtype from data, this would raise; an explicit
```
Comment (alessandrofelder) [inline src/photon_mosaic/preprocessing/tests/test_basepreprocessor.py]: Nice edge case! I think the case mentioned here currently raises with a cryptic error message (an `IndexError`) - maybe we want to catch that an return a more descriptive error message along the lines of "empty imaging objects require an explicit datatype"?
